tests/utils/fvecs_to_bin.cpp

```diff
@@ -15,11 +16,41 @@ void block_convert(std::ifstream& reader, std::ofstream& writer,
   writer.write((char*) write_buf, npts * ndims * sizeof(float));
 }
 
+// Convert byte types
+void block_convert_byte(std::ifstream& reader, std::ofstream& writer, _u8* read_buf,
+                   _u8* write_buf, _u64 npts, _u64 ndims) {
+  reader.read((char*) read_buf,
+              npts * (ndims * sizeof(_u8) + sizeof(unsigned)));
+  for (_u64 i = 0; i < npts; i++) {
+    memcpy(write_buf + i * ndims,
+           (read_buf + i * (ndims + sizeof(unsigned))) + sizeof(unsigned),
+           ndims * sizeof(_u8));
+  }
+  writer.write((char*) write_buf, npts * ndims * sizeof(_u8));
+}
+
 int main(int argc, char** argv) {
-  if (argc != 3) {
-    std::cout << argv[0] << " input_fvecs output_bin" << std::endl;
+  if (argc < 3 || argc > 4) {
+    std::cout << argv[0]
+              << " input_vecs output_bin (optional)datatype<int8/uint8/float>"
```

Comment:
Lets make datatype the first argument and mandatory to be consistent with the rest of the tools. The cmline would be <exec> datatype input.vecs output.bin.

Reply:
Changed the argument order to be <float/int8/uint8> input_vecs output_bin